scripts/prepare_test_weights.py

```diff
@@ -0,0 +1,574 @@
+      - can do a dry run to check the url is valid
+      - displays the downloaded file hash
+
+    """
+    global download_count, bytes_count
+    filename = os.path.basename(urlparse(url).path)
+    dest_filename = os.path.join(dest_folder, filename)
+    temp_filename = dest_filename + ".part"
+    dry_run = bool(os.environ.get("DRY_RUN") == "1") if dry_run is None else dry_run
+
+    is_downloaded = os.path.exists(dest_filename)
+    if is_downloaded and skip_existing:
+        skip_icon = "✖️ "
+    else:
+        skip_icon = "🔽"
+
+    if dry_run:
+        response = requests.head(url, allow_redirects=True)
+        readable_size = ""
+
+        if response.status_code == 200:
+            content_length = response.headers.get("content-length")
+
+            if content_length:
+                size_in_bytes = int(content_length)
+                readable_size = human_readable_size(size_in_bytes)
+                download_count += 1
+                bytes_count += size_in_bytes
+            print(f"✅{skip_icon} {response.status_code} READY {readable_size:<8} {url}")
+
+        else:
+            print(f"❌{skip_icon} {response.status_code} ERROR {readable_size:<8} {url}")
+        return
+
+    if skip_existing and os.path.exists(dest_filename):
+        print(f"{skip_icon}️ Skipping previously downloaded {url}")
+        return
+
+    os.makedirs(dest_folder, exist_ok=True)
+
+    print(f"🔽 Downloading {url} => '{rel(dest_filename)}'", end="\n")
+    response = requests.get(url, stream=True)
+    if response.status_code != 200:
+        print(response.content[:1000])
+        die(f"Failed to download {url}. Status code: {response.status_code}")
+    total = int(response.headers.get("content-length", 0))
+    bar = tqdm(
+        desc=filename,
+        total=total,
+        unit="iB",
+        unit_scale=True,
+        unit_divisor=1024,
+        leave=False,
+    )
+    with open(temp_filename, "wb") as f, bar:
+        for data in response.iter_content(chunk_size=1024 * 1000):
+            size = f.write(data)
+            bar.update(size)
+
+    os.rename(temp_filename, dest_filename)
+    calculated_hash = calc_hash(dest_filename)
+
+    print(f"{previous_line}✅ Downloaded {calculated_hash} {url} => '{rel(dest_filename)}' ")
+    if expected_hash is not None:
+        check_hash(dest_filename, expected_hash)
+
+
+def download_files(urls: list[str], dest_folder: str):
+    for url in urls:
+        download_file(url, dest_folder)
+
+
+def human_readable_size(size: int | float, decimal_places: int = 2) -> str:
+    for unit in ["B", "KB", "MB", "GB", "TB", "PB"]:
+        if size < 1024.0:
+            break
+        size /= 1024.0
+    return f"{size:.{decimal_places}f}{unit}"  # type: ignore
+
+
+def download_sd_text_encoder(hf_repo_id: str = "runwayml/stable-diffusion-v1-5", subdir: str = "text_encoder"):
+    encoder_filename = "model.safetensors" if "inpainting" not in hf_repo_id else "model.fp16.safetensors"
```

Comment:
Corrects filename

Reply:
True (and below too) but maybe the best fix would be not to download it at all since we never convert it. We only need the unet from the inpainting model.

Reply:
![image](https://github.com/finegrain-ai/refiners/assets/1217531/74fa0647-a9a3-4cc6-a8c2-478e1c9eac31)
Done